rfc/2022/2022-11-02_TechTeam4_Preview_Environment_Architecture.md

```diff
@@ -0,0 +1,97 @@
+1. VFS team members & VA Stakeholders without SOCKS access or the technical knowledge to use SOCKS can't use review instances
+2. Review Instance creation operates under the deprecated BRD (build, release, deploy) paradigm
+3. Review Instances don't always spin up reliably
+4. Review Instances are only available for a limited time, and therefore aren't available when needed
+5. VFS team members are unable to preview changes from multiple sources in a single preview environment
+6. VFS Teams are unclear on the benefits and intended use of Review Instances
+7. VFS team members are unable to reach review instances with tools hosted in the cloud
+8. Review Instances can't be tested on real or virtual mobile devices
+
+As a result of these issues, review instance usage is low. Review Instances are generated for every PR that is created. As of 11/18/2022, less than 10% of the review instances that are automatically created are ever accessed by a user.
+
+## Motivation
+This proposed architecture change will provide a solution to enable VFS teams to effectively preview new code and features on their products more easily and effectively, while also eliminating or mitigating all known pain points of previous solution, listed above.
+
+The new architecture, known as Preview Environments, will provide VFS teams with a capability that allows greater customization and the ability to determine for themselves what they need to effectively preview changes and collaborate with their teams.
+
+## Design
+Based on discovery, we determined there were different use cases desired from different teams. This raised the need for allowing VFS developers more control in setting up their ephemeral environment, while keeping the ability to utilize those environments as accessible as possible to facilitate internal testing, stakeholder reviews, and user testing with veterans.
+
+For the MVP of the preview environment architecture, the user will interact with GitHub normally and the preview environment will be created with default settings. As a developer pushes their branch to `vets-website` or `vets-api`, the preview environment architecture will detect the branch and begin the process of deploying the code in the branch to a preview environment. Once the continuous integration workflow completes successfully, a request to deploy the preview environment triggers a GitHub Actions workflow. The GHA workflow writes the dynamically generated preview environment configuration to the `app-manifest` repository. EKS detects the change in the `devops` repository and deploys the newly defined environment.
+
+A diagram is submitted [below](#Diagrams) as additional explanation.
+
+**MVP Feature Set:**
+1. Preview Environment MVP will create ephemeral environments that can be accessed without SOCKS
+2. Preview Environment MVP will operate on the EKS infrastructure and paradigm
+3. Preview Environment MVP will automatically create a Preview Environment when a PR is created
```

Comment:
Could we have some logic here, key phrase in the PR name or something that gets detected to do a PE? This way we can narrow in on actual desire to use the capability as well as keeping resource use, etc. down.

Reply:
Yeah, this idea is also on our radar. One thought offered by Kyle I believe, was to tag a pull request with a specific label indicating the need to have the PR deployed that we detect in the workflow.